internal/uvm/start.go

```diff
@@ -115,6 +118,38 @@ func parseLogrus(vmid string) func(r io.Reader) {
 	}
 }
 
+// When using an external GCS connection it is necessary to send a ModifySettings request
+// for HvSockt so that the GCS can setup some registry keys that are required for running
+// containers inside the UVM. Usually this is done by the HCS immediately after the GCS
+// connection is done. Since, we are using the external GCS connection we should do that
+// setup here after we connect with the GCS.
+// This only applies for WCOW
+func (uvm *UtilityVM) configureHvSocketForGCS(ctx context.Context) (err error) {
+	if uvm.OS() != "windows" {
+		return nil
+	}
+
+	hvsocketAddress := &guestrequest.HvSocketAddress{
+		LocalAddress:  uvm.runtimeID.String(),
+		ParentAddress: gcs.WindowsGcsHvHostID.String(),
+	}
+
+	conSetupReq := &hcsschema.ModifySettingRequest{
+		RequestType: requesttype.Update,
+		GuestRequest: guestrequest.GuestRequest{
+			RequestType:  requesttype.Update,
+			ResourceType: guestrequest.ResourceTypeHvSocket,
+			Settings:     hvsocketAddress,
+		},
+	}
+
```

Comment:
I'm still a little confused as to what is being updated? What exists already?

Reply:
Seconded kind of, does this setup registry keys solely related to the hvsocket addr?

Reply:
This code is ported as it is from the HCS code. (Look at VirtualMachineOcrchestrator.cpp:1339). If this setup is not done immediately after the connection is established then we will keep getting the `The group or resource is not in the correct state to perform the requested operation` error. I found out about this from Kris Harper. So even I don't know exactly what registry keys are being setup by this call.

Reply:
Ok thank you for the clarification :)